crates/cast/src/lib.rs

```diff
@@ -1824,6 +1824,17 @@ impl SimpleCast {
             None => eyre::bail!("No selector found"),
         }
     }
+
+    pub fn decode_tx(tx: String) -> Result<(String, String)> {
+        let tx_hex = hex::decode(strip_0x(&tx))?;
+        let tx_rlp = rlp::Rlp::new(tx_hex.as_slice());
+        let (tx, sig) = TypedTransaction::decode_signed(&tx_rlp)?;
+
+        let tx = serde_json::to_string_pretty(&tx)?;
+        let sig = serde_json::to_string_pretty(&sig)?;
```

Comment:
this is usually less than ideal, because this effectively renders this function useless for any other use case than printing.

the function should instead just decode

the command impl can then convert to string